docs/user/user/pings/index.md

```diff
@@ -95,21 +225,91 @@ where:
+Glean metric [`glean.upload.discarded_exceeding_pings_size`](../collected-metrics/metrics.md#metrics-1).
+
+#### Storage quota
+
+Pending pings are stored on disk. Storage is scanned every time Glean is initialized and upon scanning
+Glean checks its size. If it exceeds a size of 10MB or 250 pending ping, pings are deleted to get the
+storage back to an accepted size. Pings are deleted oldest first, until the storage size is below the quota.
+
+The number of deleted pings due to exceeding storage quota is recorded on the metric
+[`glean.upload.deleted_pings_after_quota_hit`](../collected-metrics/metrics.md#metrics-1)
+and the size of the pending pings directory is recorded (regardless on whether quota has been reached)
+on the metric [`glean.upload.pending_pings_directory_size`](../collected-metrics/metrics.md#metrics-1).
+
+Deletion request pings are not subject to this limitation and never get deleted.
 
 ### Submitted headers
-A pre-defined set of headers is additionally sent along with the submitted ping:
-
-| Header | Value | Description |
-|--------|-------|-------------|
-| `Content-Type` | `application/json; charset=utf-8` | Describes the data sent to the server |
-| `User-Agent` | Defaults to e.g. `Glean/0.40.0 (Kotlin on Android)`, where `0.40.0` is the Glean SDK version number and `Kotlin on Android` is the name of the language used by the binding that sent the request plus the name of the platform it is running on. | Describes the application sending the ping using the Glean SDK |
-| `Date` | e.g. `Mon, 23 Jan 2019 10:10:10 GMT+00:00` | Submission date/time in GMT/UTC+0 offset |
-| `X-Client-Type` | `Glean` | Custom header to support handling of Glean pings in the legacy pipeline |
-| `X-Client-Version` | e.g. `0.40.0` | The Glean SDK version, sent as a custom header to support handling of Glean pings in the legacy pipeline |
-| `X-Debug-ID` | *Optional*, e.g. `test-tag` | Debug header attached to Glean pings when using the [debug tools](../../user/debugging/index.md) |
-| `X-Source-Tags` | *Optional*, e.g. `automation, perf` | A list of tags to associate with the ping, useful for clustering pings at analysis time, for example to tell data generated from CI from other data. |
+
+A pre-defined set of headers is additionally sent along with the submitted ping.
+
+#### `Content-Type`
+
+Describes the data sent to the server. Value is always `application/json; charset=utf-8`.
+
+#### `Date`
+
+Submission date/time in GMT/UTC+0 offset e.g. `Mon, 23 Jan 2019 10:10:10 GMT+00:00`.
+
+#### `User-Agent`
+
+The Glean SDKs do not set this header[^1], so it will contain whatever value was set
+by the underlying uploading mechanism. For example, when sending pings from browsers
+it will contain the characteristic [browser UA string](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent#syntax).
+
+This header is parsed by the Glean pipeline and can be queried at analysis time through
```

Comment:
Might be good to call out that in old (actually current) Glean versions this is the same as `X-Telemetry-Agent` in newer ones (at least for Glean core)

Reply:
I did call it out in the footnote https://github.com/mozilla/glean/pull/1744/files#diff-7941a3cb068acb37c109d8f78146a0b31b52b984271ed3767130a61a0fccb1f8R312-R315

Do you think that is too hidden?

Reply:
Should be fine when rendered then.